### PR TITLE
(feat) Poller behaviour with select2 MON-10870

### PR DESCRIPTION
**Pull request by tanguyvda**
The PDO request try to bind a string of multiple id separated with comma, instead of binding each id
+ adds poller acl into the select2 list
this is mostly a rip-off of #7119

# Pull Request Template

## Description

this PR is needed for this PR: https://github.com/centreon/centreon-widget-engine-status/pull/22

* **QA Team** (Quality Assurance) with tests.
* **reviewers** to understand what are the stakes of the pull request.

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [x] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [x] 18.10.x
- [x] 19.04.x
- [x] 19.10.x (master)

<h2> How this pull request can be tested ? </h2>
you need to use the engine-status widget with this PR: 
https://github.com/centreon/centreon-widget-engine-status/pull/22

when you install this widget with the aformentioned PR, you're going to be able to filter on one or more host categories. Select at least two of them in the parameters of your widget.

Save your parameters.
get back to the widget parameters, you'll have a PDO issue and all your categories will have vanished.

With this PR, they won't vanish anymore. @sc979 knows what I'm talking about ;)

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [x] I followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

**Review by kduret — DISMISSED** [review]

File: www/class/centreonInstance.class.php [hunk 1]
```diff
@@ -215,20 +215,31 @@ public function setCommands($pollerId, $commands)
...
+        global $centreon;
+
         $selectedInstances = '';
         $items = array();
+
+        if (empty($values)) {
+            return $items;
+        }
+
+        # get list of authorized pollers
```
Comment (loiclau) [inline www/class/centreonInstance.class.php]:
```suggestion
        // get list of authorized pollers
```

**Review by loiclau — DISMISSED** [review]

File: www/class/centreonInstance.class.php (hunk 1 above)
Comment (tanguyvda) [reply www/class/centreonInstance.class.php]: bash is still my first love ^^'

**Comment by omercier** [comment]
#MON-10870

**Review by TamazC — DISMISSED** [review]

**Review by kduret — APPROVED** [review]

**Comment by kduret** [comment]
replaced by https://github.com/centreon/centreon/pull/11660

**Pull request closed without merging**